rust-runtime/aws-smithy-runtime-api/src/client/identity.rs

```diff
@@ -161,19 +161,45 @@ pub trait ResolveIdentity: Send + Sync + Debug {
     }
 }
 
+/// Cache location for identity caching.
+///
+/// Identities are usually cached in the identity cache owned by [`RuntimeComponents`]. However,
+/// we do have identities whose caching mechanism is internally managed by their identity resolver,
+/// in which case we want to avoid the `RuntimeComponents`-owned identity cache interfering with
+/// the internal caching policy.
+#[non_exhaustive]
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum IdentityCacheLocation {
+    /// Indicates the identity cache is owned by [`RuntimeComponents`].
+    RuntimeComponents,
+    /// Indicates the identity cache is internally managed by the identity resolver.
+    IdentityResolver,
+}
+
 /// Container for a shared identity resolver.
 #[derive(Clone, Debug)]
 pub struct SharedIdentityResolver {
     inner: Arc<dyn ResolveIdentity>,
     cache_partition: IdentityCachePartition,
+    cache_location: IdentityCacheLocation,
```

Comment:
Shouldn't this defer to the wrapped `inner` identity resolver trait impl?

Reply:
Just wanna make sure you read the first white sub-bullet under `Implementation decisions` that says `Two reasons why CacheLocation is added to SharedIdentityResolver...` and then still have this question?

Reply:
Addressed in c2a2708